src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -810,6 +811,17 @@ public IMethodSymbol GetEntryPoint(CancellationToken cancellationToken)
             return (INamedTypeSymbol?)CommonGetSpecialType(specialType)?.GetITypeSymbol();
         }
 
+        public INamedTypeSymbol GetSpecialTypeOrThrow(SpecialType specialType)
+        {
+            var symbol = GetSpecialType(specialType);
+            if (symbol is null)
+            {
+                throw new InvalidOperationException($"Special type {specialType} is not available.");
```

Comment:
If the method is `public`, should this be a string resource?

Reply:
Maybe I should be using `ExceptionUtilities.Unreachable` here. The expectation is that this is never hit. The caller is assumed to have validated that the type is already available.

---
In reply to: [360000477](https://github.com/dotnet/roslyn/pull/40493#discussion_r360000477) [](ancestors = 360000477)

Reply:
What method would the caller use to validate the type exists other than `GetSpecialType()`?

---
In reply to: [360001353](https://github.com/dotnet/roslyn/pull/40493#discussion_r360001353) [](ancestors = 360001353,360000477)

Reply:
In several cases the IDE code has a symbol for `Task` in hand and they assume that types like `void` are available. After all if `void` isn't available we'll not get to the point of binding `Task`.

Reply:
@jaredpar: I think Unreachable is only really used for internal places where we don't think it can ever be reached. An analyzer author writing bad code shouldn't see that.